File: two-pointers/remove-duplicates-from-sorted-array.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        int len = std::size(nums);
        int pointer = 1; 
        int temp = nums[1];
        for (int i = 1; i < len; i ++){
            if (nums[i] != temp) {
                nums[pointer] = nums[i];
                pointer ++;
                int temp = nums[i];
            }
        }
        return pointer;
    }
};

File: two-pointers/remove-duplicates-from-sorted-array.py
class Solution(object):
    def removeDuplicates(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """

        pointer = 0
        temp = None

        for i in range(len(nums)):
            if nums[i] != temp:
                temp = nums[i]
                nums[pointer] = nums[i]
                pointer += 1

        
        return pointer